core/src/main/java/io/ultrabrew/metrics/data/BasicHistogramAggregator.java

```diff
@@ -0,0 +1,204 @@
+package io.ultrabrew.metrics.data;
+
+import io.ultrabrew.metrics.util.DistributionBucket;
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.Unsafe;
+
+public class BasicHistogramAggregator extends ConcurrentIntTable implements Aggregator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BasicHistogramAggregator.class);
+  private static final String[] AGG_FIELDS = {"count", "sum", "min", "max"};
+  private static final int[] IDENTITY = {0, 0, Integer.MAX_VALUE, Integer.MIN_VALUE};
+
+  private final String metricId;
+  private final DistributionBucket buckets;
+  private final int bucketCount;
+
+  public BasicHistogramAggregator(final String metricId, final DistributionBucket buckets) {
+    this(metricId, buckets, DEFAULT_CAPACITY, DEFAULT_MAX_CAPACITY);
+  }
+
+  public BasicHistogramAggregator(final String metricId, final DistributionBucket buckets,
+      final int capacity) {
+    this(metricId, buckets, capacity, DEFAULT_MAX_CAPACITY);
+  }
+
+  public BasicHistogramAggregator(final String metricId, final DistributionBucket buckets,
+      final int capacity, final int maxCapacity) {
+    super(AGG_FIELDS.length + buckets.getCount(), capacity, maxCapacity);
+    this.metricId = metricId;
+    this.buckets = buckets;
+    this.bucketCount = buckets.getCount();
+  }
+
+  @Override
+  public void apply(String[] tags, long latency, long timestamp) {
+    apply(tags, (int) latency, timestamp);
+  }
+
+  @Override
+  protected void combine(int[] table, long baseOffset, int latency) {
+    add(table, baseOffset, 0, 1);
```

Comment:
if we keep the base array type as `int[]` we need to write a long into 2 `int`s. need to adjust the offsets accordingly.